pkg/image/validation/kubernetes.go

```diff
@@ -317,9 +323,51 @@ func validateHelmRepoArgs(parsedURL *url.URL, repo *image.HelmRepository) []Fail
 		})
 	}
 
+	if repo.SkipTLSVerify && repo.CAFile != "" {
+		failures = append(failures, FailedValidation{
+			UserMessage: fmt.Sprintf("Helm repository 'caFile' field for %q should not be defined while 'skipTLSVerify' is true.", repo.Name),
+		})
+	}
+
+	if repo.PlainHTTP && repo.CAFile != "" {
+		failures = append(failures, FailedValidation{
+			UserMessage: fmt.Sprintf("Helm repository 'caFile' field for %q should not be defined while 'plainHTTP' is true.", repo.Name),
+		})
+	}
+
+	if parsedURL.Scheme == httpScheme && repo.CAFile != "" {
+		failures = append(failures, FailedValidation{
+			UserMessage: fmt.Sprintf("Helm repository 'url' field for %q contains 'http://' but 'caFile' field is defined.", repo.Name),
+		})
+	}
+
 	return failures
 }
 
+func validateHelmRepoCert(repoName, certFile string, imageConfigDir string) string {
+	if certFile == "" {
+		return ""
+	}
+
+	validExtensions := []string{".pem", ".crt", ".cer", ".der", ".p7b", ".p7c", ".pfx", ".p12"}
```

Comment:
@rdoxenham could you please confirm these?

Reply:
We're only talking about CA bundles here, I'm afraid I don't know enough on the topic. 

```
      --ca-file string     verify certificates of HTTPS-enabled servers using this CA bundle
      --cert-file string   identify registry client using this SSL certificate file
```

Reply:
Should note that `--cert-file` file isn't included in this, this would just be support for `--ca-file`. `--cert-file` is a separate thing that needs to be used in combination with `--key-file` (as far as I understand).